src/server/server_state.h

```diff
@@ -15,10 +15,76 @@ typedef struct mi_heap_s mi_heap_t;
 
 namespace dfly {
 
+class ConnectionContext;
 namespace journal {
 class Journal;
 }  // namespace journal
 
+// This would be used as a thread local storage of sending
+// monitor messages.
+// Each thread will have its own list of all the connections that are
+// used for monitoring. When a connection is set to monitor it would register
+// itself to this list on all i/o threads. When a new command is dispatched,
+// and this list is not empty, it would send in the same thread context as then
+// thread that registered here the command.
+// Note about performance: we are assuming that we would not have many connections
+// that are registered here. This is not pub sub where it must be high performance
+// and may support many to many with tens or more of connections. It is assumed that
+// since monitoring is for debugging only, we would have less than 1 in most cases.
+// Also note that we holding this list on the thread level since this is the context
+// at which this would run. It also minimized the number of copied for this list.
+class MonitorsRepo {
+ public:
+  struct MonitorInfo {
```

Comment:
i think this struct is redundant now and, in any case it seems like an implementation detail - `Add` should accept ConnectionContext similarly to `Remove`

Reply:
removed